enforcer/src/main/java/org/wso2/choreo/connect/enforcer/server/WebSocketHandler.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.choreo.connect.enforcer.server;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.choreo.connect.discovery.service.websocket.WebSocketFrameRequest;
+import org.wso2.choreo.connect.enforcer.api.APIFactory;
+import org.wso2.choreo.connect.enforcer.api.RequestContext;
+import org.wso2.choreo.connect.enforcer.api.WebSocketAPI;
+import org.wso2.choreo.connect.enforcer.api.config.APIConfig;
+import org.wso2.choreo.connect.enforcer.constants.APIConstants;
+import org.wso2.choreo.connect.enforcer.security.AuthenticationContext;
+import org.wso2.choreo.connect.enforcer.websocket.MetadataConstants;
+import org.wso2.choreo.connect.enforcer.websocket.WebSocketFrameContext;
+import org.wso2.choreo.connect.enforcer.websocket.WebSocketThrottleResponse;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+/**
+ * WebSocketHandler handles requests coming through websocket frame service.
+ */
+public class WebSocketHandler implements RequestHandler<WebSocketFrameRequest, WebSocketThrottleResponse> {
+    private static final Logger logger = LogManager.getLogger(WebSocketHandler.class);
+
+    /**
+     *
+     * @param webSocketFrameRequest - WebSocketFrameRequest received from WebSocketFrameRequest
+     * @return WebSocketResponseObject - Response sent via WebSocketFrameService
+     */
+    @Override
+    public WebSocketThrottleResponse process(WebSocketFrameRequest webSocketFrameRequest) {
+        WebSocketAPI matchedAPI = APIFactory.getInstance().getMatchedAPI(webSocketFrameRequest);
+        if (matchedAPI == null) {
```

Comment:
What is the response retuned in this case. In rest API we create response object in here

Reply:
`WebSocketThrottleResponse` contains the current throttle state and the UTC time in seconds for throttle out time.

Reply:
Shouldn't we log a message if the matched API is null?

Reply:
fixed